### PR TITLE
Add CONTRIBUTING file and pre-commit hook instructions

### PR DESCRIPTION
**Pull request by flferretti**
This pull request adds a CONTRIBUTING file to provide guidelines for contributing to the project. It also includes instructions for setting up and using the pre-commit hook to ensure code quality before committing.

----
📚 Documentation preview 📚: https://jaxsim--87.org.readthedocs.build//87/

### CODE REVIEW

**Comment by traversaro** [comment]
I guess we can add the `.md` extension to `CONTRIBUTING` ? So it gets rendered correctly.

**Comment by flferretti** [comment]
> I guess we can add the `.md` extension to `CONTRIBUTING` ? So it gets rendered correctly.

Done in https://github.com/ami-iit/jaxsim/pull/87/commits/6fd42f24213046b5f038b0aa642f07c10a8116d0

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+To manually run the pre-commit hook at any time, use:
+
+```bash
+pre-commit run --all-files
+```
+
+## Branching Strategy :deciduous_tree:
+
+When making changes, create a new branch from main. Use a descriptive name like `feature/your-feature` or `fix/your-fix`. This helps us understand the purpose of the branch.
```
Comment (diegoferigo) [inline CONTRIBUTING.md:41]: The project is r/o for external contributors, I would either remove this section or probably suggest to create a branch in their own fork.

**Review by diegoferigo — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (flferretti) [reply CONTRIBUTING.md:41]: Done in abc4eb479933aa1ed78fb6d4389b487804312e2b

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+Your code will be tested with the CI/CD pipeline before merging. Feel free to add new ones or update the existing tests in the [workflows](./github/workflows) folder to cover your changes.
+
+## Documentation :book:
+
+Update the documentation in the [docs](./docs) folder and the [README](./README.md) to reflect your changes, if necessary.
+
+## Code Review :eyes:
+
+Expect feedback during the code review process. Address comments and make necessary changes. This collaboration ensures quality.
+
```
[the remark this answers is not in the document]
Comment (flferretti) [reply CONTRIBUTING.md:58]: Done in aa12d193d9192d3a7b5bf66e941359e02bdb5c6e

**Review by traversaro — APPROVED** [review]

**Comment by flferretti** [comment]
Thanks @diegoferigo and @traversaro! I will squash and merge this

**Pull request merged**